src/main/java/com/auth0/net/SignUpRequest.java

```diff
@@ -1,11 +1,14 @@
 package com.auth0.net;
 
+import com.auth0.json.auth.UserInfo;
+
 import java.util.Map;
 
 /**
  * Class that represents a Create User call.
  */
-public interface SignUpRequest extends Request<Void> {
+@SuppressWarnings("UnusedReturnValue")
+public interface SignUpRequest extends Request<UserInfo> {
```

Comment:
So previously the sign up did work. However it would simply return a void response on success.
So a developer may be checking for `null` to detect success? Which would now be a BC as the response is not null?

Reply:
`Void` [class](https://docs.oracle.com/javase/7/docs/api/java/lang/Void.html#TYPE) is used in java generics as you can't return "void" nor a primitive value, they must actually return an object. When a method says it returns "Void" it must return the "null value" as seen [here](https://github.com/auth0/auth0-java/blob/master/src/main/java/com/auth0/net/VoidRequest.java#L22).

If they wanted to check success they would just try/catch the `request.execute()` call. But holding the `Void` instance is nonsense as they can't do anything with that. So probably, everyone calling this method before these changes would have something like this:

```java
AuthAPI authAPI = new AuthAPI("domain.auth0.com", "clientId", "myClientSecret");
SignUpRequest signUpRequest = authAPI.signUp("email@admin.com", "pwd123", "connect1on");
try {
    signUpRequest.execute();
} catch (Auth0Exception e) {
    //Check errors
    e.printStackTrace();
}
```

and after the change they CAN add the return type (the snippet above will continue working just fine as well👌 ).

```java
AuthAPI authAPI = new AuthAPI("domain.auth0.com", "clientId", "myClientSecret");
SignUpRequest signUpRequest = authAPI.signUp("email@admin.com", "pwd123", "connect1on");
try {
    UserInfo info = signUpRequest.execute();
    //use "info"
} catch (Auth0Exception e) {
    //Check errors
    e.printStackTrace();
}
```

Yes, previous method did work but the response wasn't the right one. Is there an alternative? The only breaking change I can consider is if they were using the nonsense version of the snippet:

```java
AuthAPI authAPI = new AuthAPI("domain.auth0.com", "clientId", "myClientSecret");
SignUpRequest signUpRequest = authAPI.signUp("email@admin.com", "pwd123", "connect1on");
try {
    Void result = signUpRequest.execute(); // --> Compile error as Void is now UserInfo
} catch (Auth0Exception e) {
    //Check errors
    e.printStackTrace();
}
```

 They'd need to fix the type of the returned object.